tst/testinstall/methwhy.tst

```diff
@@ -91,124 +136,207 @@ function ( x, y... )
-function ( x, y... )
-    return 1;
-end
-gap> Display(ApplicableMethod(foobar, [1..3], 5));
+#I  Method 2, applicable method number 1, value: 0
+#I  ``foobar: for 2 integers''
+#I   at stream:2
+function( x, y... ) ... end
+gap> ApplicableMethod(foobar, [1..3], 4);
 #I  Searching Method for foobar with 3 arguments:
-#I  Total: 1 entries
-#I  Method 1: ``foobar: for 3 integers'' at stream:3, value: 0
-function ( x, y... )
-    return 1;
-end
-gap> Display(ApplicableMethod(foobar, [1..4], 5));
+#I  Total: 1 entries, of which 1 is applicable:
+#I  Method 1, applicable method number 1, value: 0
+#I  ``foobar: for 3 integers''
+#I   at stream:3
+function( x, y... ) ... end
+gap> ApplicableMethod(foobar, [1..4], 4);
 #I  Searching Method for foobar with 4 arguments:
-#I  Total: 1 entries
-#I  Method 1: ``foobar: for 4 integers'' at stream:4, value: 0
-function ( x, y... )
-    return 1;
-end
-gap> Display(ApplicableMethod(foobar, [1..5], 5));
+#I  Total: 1 entries, of which 1 is applicable:
+#I  Method 1, applicable method number 1, value: 0
+#I  ``foobar: for 4 integers''
+#I   at stream:4
+function( x, y... ) ... end
+gap> ApplicableMethod(foobar, [1..5], 2);
 #I  Searching Method for foobar with 5 arguments:
-#I  Total: 1 entries
-#I  Method 1: ``foobar: for 5 integers'' at stream:5, value: 0
-function ( x, y... )
-    return 1;
-end
-gap> Display(ApplicableMethod(foobar, [1..6], 5));
+#I  Total: 1 entries, of which 1 is applicable:
+#I  Method 1, applicable method number 1, value: 0
+#I  ``foobar: for 5 integers''
+#I   at stream:5
+function( x, y... ) ... end
+gap> ApplicableMethod(foobar, [1..6], 1);
 #I  Searching Method for foobar with 6 arguments:
-#I  Total: 1 entries
-#I  Method 1: ``foobar: for 6 integers'' at stream:6, value: 0
-function ( x, y... )
-    return 1;
+#I  Total: 1 entries, of which 1 is applicable:
+#I  Method 1, applicable method number 1, value: 0
+#I  ``foobar: for 6 integers''
+#I   at stream:6
+function( x, y... ) ... end
+gap> ApplicableMethod(foobar, [1..7], 1);
+#I  no method found for this operation with this number of arguments
+fail
+
+#
+# check ApplicableMethod when applied to a function 
+#
+gap> foobar := function(a,b,c) return 1; end;; 
+gap> ApplicableMethod( foobar, [4,5,6], 1, 1 );       
+#I  foobar is a function, not an operation
+#I  and is located at: stream:1
+function( a, b, c ) ... end
+gap> ApplicableMethod( foobar, [7,8], 1, 1 );  
+#I  foobar is a function, not an operation
+#I  and requires 3 arguments
+fail
+gap> foobar := function(x,arg...) return x*arg[1]; end;;
+gap> ApplicableMethod( foobar, [7,8], 1, 1 );           
+#I  foobar is a function, not an operation
+#I  and is located at: stream:1
+function( x, arg... ) ... end
+gap> ApplicableMethod( foobar, [9], 1, 1 );  
+#I  foobar is a function, not an operation
+#I  and requires at least 2 arguments
+fail
+
+#
+# now not returning 'fail' when the second argument fails to be a list, 
+# but replacing arg2 by [arg2] and then carrying on:
```

Comment:
I am conflicted about this. While I've made the mistake myself in the past, it usually was then trivial to correct. It'd be easier if there was a good error message, of course ("<args> must be a list, not an integer/a boolean/a ..."). But by allowing it, we allow users to learn that this is OK. But once they accidentally write `list` instead of `[list]`, they again get a wrong output; but now we are again in a potentially confusing and hard to recover from situation.

So I think I personally would prefer if this gave a helpful error, instead of accepting the wrong input in some cases. Anyway, I certainly won't veto this PR over this minor bit.